applications/DEMApplication/python_scripts/DEM_procedures.py

```diff
@@ -608,9 +610,10 @@ def AddClusterVariables(self, model_part, DEM_parameters):
             model_part.AddNodalSolutionStepVariable(EULER_ANGLES)
 
         # CONTROL MODULE
-        model_part.AddNodalSolutionStepVariable(TARGET_STRESS)
-        model_part.AddNodalSolutionStepVariable(REACTION_STRESS)
-        model_part.AddNodalSolutionStepVariable(LOADING_VELOCITY)
+        if DEM_parameters["PostControlModule"].GetBool():
+            model_part.AddNodalSolutionStepVariable(TARGET_STRESS)
+            model_part.AddNodalSolutionStepVariable(REACTION_STRESS)
+            model_part.AddNodalSolutionStepVariable(LOADING_VELOCITY)
```

Comment:
I agree that control module variables should only be added when somebody uses the control module, but we need to think of a cleaner way to do it. With that if statement, the usage of control module depends on whether you print the control module variables or not, which is not the most appropriate (you could perfectly be using the control module without printing those variables).